t/output-filter-group-collection.t

```diff
@@ -45,6 +45,19 @@ given Agrammon::Outputs::FilterGroupCollection.from-scalar(0) {
     }
 }
 
+{
+    my $group = Agrammon::Outputs::FilterGroupCollection.from-filter-to-value-pairs:
+            [{ac => 'blue cow'} => 4, {ac => 'pink cow'} => 7, {ac => 'blue cow'} => 31];
+    given $group.add(2) {
+        isa-ok $_, Agrammon::Outputs::FilterGroupCollection,
+                'Get another filter group back after adding scalar';
+        is +$group, 42, 'Original filter group is not changed in place';
+        is +$_, 48, 'Total numeric value is correct after adding scalar';
+        is-deeply norm(.results-by-filter-group),
```

Comment:
I think this should be 46, as there are only two different keys in the filter group (`blue cow` and `pink cow`), thus 2 is only added twice.

Reply:
I thought that there's no filter grouping when applying such a function. Line 44 says: `'Correct results by filter group after scaling';`, so I guessed there's the scaling and then an extra call to `.results-by-filter-group`? Same for `add`...

Reply:
I'm going to change the test to `[{ac => 'blue cow'} => 4, {ac => 'pink cow'} => 7, {ac => 'green cow'} => 31];`, since we never can have a situation where we use add and the filter group contains two identical keys, anyway.

Reply:
Fixed in 67d5340fb2593e23b05264e1cfc60459320f99b8

Reply:
The filter groups main purpose is to add up the values with identical keys (for example, animal categories).
The functions dealing with filter groups **must** maintain this property, otherwise their behaviour will be inconsistent.
Therefore, the test for 

> I thought that there's no filter grouping when applying such a function. Line 44 says: `'Correct results by filter group after scaling';`, so I guessed there's the scaling and then an extra call to `.results-by-filter-group`? Same for `add`...

The sentence you quote describes what a successful test will be. The `scale()` function works on an existing filter group, so the order of evaluation is
1. group by filter key
2. multiply the values with the factor for each key.

Reply:
What @zaucker says is accurate. I guess addition of a scalar to a filter group can be useful for something (otherwise, why would you be adding it :-)), but unlike scaling, its semantics are not so neat. With scaling, the grouping of keys does not affect the result, and if you scaled the total you'd get the same outcome. Addition is rather different; I guess if you know your input data and know what result you want, it's fine, but so far as the model of "filter groups are just a value with a breakdown" goes, addition to each key doesn't fit.